### DISCUSSION TITLE
Polars support for Great Expectations

### DISCUSSION THREAD

**Discussion by Durga854** [great-expectations/great_expectations#10144] [Ideas]
Hello,
Could you please consider adding support for Polars in Great Expectations? It would be great to have functionality similar to what is available for Pandas and Spark. Thanks for your attention to this!

**Comment by matthiasgomolka** [comment]
I support this. Pandas is often quite slow.

Maybe it would be helpful to use [narwhals](https://pypi.org/project/narwhals/) in this process. Or maybe even [ibis](https://pypi.org/project/ibis-framework/) ...

**Reply by MarcoGorelli** [reply]
[Narwhals](https://github.com/narwhals-dev/narwhals) is actually already a requirement of great-expectations, because great-expectations requires Altair which in turn requires Narwhals, so it could be used without introducing any extra dependencies 😉 

Happy to help if the maintainers are open to this, feel free to [reach out on Discord](https://discord.gg/V3PqtB4VA4) if you're interested

**Reply by Kilo59** [reply]
@MarcoGorelli 

This is super interesting.
The first step might be to copy the [PandasExecutionEngine](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/execution_engine/pandas_execution_engine.py) and create a `NarwhalsExecutionEngine`. At that point, it should be trivial to create a new `PolarsDatasource` type that simply uses the `NarwhalsExecutionEngine` instead of the pandas one.

A specific `PolarsExecutionEngine` could also be created directly, but I suspect that might be a bit more work.

**Reply by Kilo59** [reply]
There seems to be a fair amount of interest in this.
- See https://github.com/great-expectations/great_expectations/issues/10702
